bulldozer/merge.go

```diff
@@ -149,21 +149,48 @@ func (m *PushRestrictionMerger) DeleteHead(ctx context.Context, pullCtx pull.Con
 	return m.Normal.DeleteHead(ctx, pullCtx)
 }
 
-// MergePR merges a pull request if all conditions are met. It logs any errors
-// that it encounters.
-func MergePR(ctx context.Context, pullCtx pull.Context, merger Merger, mergeConfig MergeConfig) {
+// DetermineMergeMethod determines which merge method to use when merging the PR
+func DetermineMergeMethod(ctx context.Context, pullCtx pull.Context, mergeConfig MergeConfig) (MergeMethod, error) {
 	logger := zerolog.Ctx(ctx)
 
-	base, head := pullCtx.Branches()
+	base, _ := pullCtx.Branches()
 	mergeMethod := mergeConfig.Method
 
 	if branchMergeMethod, ok := mergeConfig.BranchMethod[base]; ok {
 		mergeMethod = branchMergeMethod
 	}
+
+	for _, method := range mergeConfig.MergeMethods {
+		triggered, reason, err := IsMergeMethodTriggered(ctx, pullCtx, method.Trigger)
+		if err != nil {
+			logger.Error().Err(err).Msg("Failed to determine if merge method is triggered")
+			return "", err
```

Comment:
Since this returns an error, we should probably log it in the caller, which does not return an error, instead of here.